app/components/custom-fields/form.tsx

```diff
@@ -0,0 +1,153 @@
+import { CustomFieldType, type CustomField } from "@prisma/client";
+import { Form, useNavigation } from "@remix-run/react";
+import { useAtom } from "jotai";
+import { useZorm } from "react-zorm";
+import { z } from "zod";
+import { updateTitleAtom } from "~/atoms/custom-fields.new";
+import { useOrganizationId } from "~/hooks/use-organization-id";
+import { isFormProcessing } from "~/utils";
+import FormRow from "../forms/form-row";
+import Input from "../forms/input";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../forms/select";
+import { Switch } from "../forms/switch";
+import { Button } from "../shared";
+import { Spinner } from "../shared/spinner";
+
+export const NewCustomFieldFormSchema = z.object({
+  name: z.string().min(2, "Name is required"),
+  helpText: z
+    .string()
+    .optional()
+    .transform((val) => val || null), // Transforming undefined to fit prismas null constraint
+  type: z.nativeEnum(CustomFieldType),
+  required: z
+    .string()
+    .optional()
+    .transform((val) => (val === "on" ? true : false)),
+  organizationId: z.string(),
+});
+
+/** Pass props of the values to be used as default for the form fields */
+interface Props {
+  name?: CustomField["name"];
+  helpText?: CustomField["helpText"];
+  required?: CustomField["required"];
+}
+
+export const CustomFieldForm = ({ name, helpText, required }: Props) => {
+  const navigation = useNavigation();
+  const zo = useZorm("NewQuestionWizardScreen", NewCustomFieldFormSchema);
+  const disabled = isFormProcessing(navigation.state);
+  const fieldTypes = ["Text"];
+
+  const [, updateTitle] = useAtom(updateTitleAtom);
+
+  const organizationId = useOrganizationId();
+  return (
+    <Form
```

Comment:
Submitting with cmd+enter doesn't seem to work and it works on example on asset.new. Can you please check why?